discord-commands/src/test/java/com/blank/humanity/discordbot/services/ShopServiceTest.java

```diff
@@ -0,0 +1,222 @@
+
+        int available = shopService.getAvailableItemAmount(item);
+        assertThat(available).isZero();
+    }
+
+    @Test
+    void testAvailableItemAmountUnlimited() {
+        ShopItem item = itemShopConfig.getShopItem(NONE_BOUGHT_SHOP_ID).get();
+
+        int available = shopService.getAvailableItemAmount(item);
+        assertThat(available).isEqualTo(-1);
+    }
+
+    @Test
+    void testGetShopPage() {
+        List<ShopItem> items = shopService.getShopPage(2);
+
+        assertThat(items).isNotNull().doesNotContainNull().hasSize(2);
+        assertThat(items.get(0).getItemId()).isEqualTo(3);
+        assertThat(items.get(1).getItemId()).isEqualTo(2);
+    }
+
+    @Test
+    void testAmountShopPages() {
+        long shouldBe = (long) Math.ceil(SHOP_ITEMS / ITEMS_PER_PAGE);
+        assertThat(shopService.amountShopPages()).isEqualTo(shouldBe);
+    }
+
+    @Test
+    void testBuyItemSingle() throws NotEnoughBalanceException {
+        BlankUser user = new BlankUser();
+        user.setBalance(1234);
+        user.setId(1l);
+
+        ShopItem item = itemShopConfig.getShopItem(NONE_BOUGHT_SHOP_ID).get();
+
+        ItemBuyStatus status = shopService.buyItem(user, item);
+        assertThat(status).isEqualTo(ItemBuyStatus.SUCCESS);
+
+        verify(blankUserService).decreaseUserBalance(user, item.getPrice());
+        verify(inventoryService).giveItem(user, item.getItemId(), 1);
+
+        assertThat(buyLogDao.sumOfBoughtItems(item.getId())).isEqualTo(1);
+    }
+
+    @Test
+    void testBuyItemMultiple() throws NotEnoughBalanceException {
+        int amount = 6;
+
+        BlankUser user = new BlankUser();
+        user.setBalance(1234);
+        user.setId(1l);
+
+        ShopItem item = itemShopConfig.getShopItem(NONE_BOUGHT_SHOP_ID).get();
+
+        ItemBuyStatus status = shopService.buyItem(user, item, amount);
+        assertThat(status).isEqualTo(ItemBuyStatus.SUCCESS);
+
+        verify(blankUserService)
+            .decreaseUserBalance(user, amount * item.getPrice());
+        verify(inventoryService).giveItem(user, item.getItemId(), amount);
+
+        assertThat(buyLogDao.sumOfBoughtItems(item.getId())).isEqualTo(amount);
+    }
+
+    @Test
+    void testBuyItemNotEnoughBalance() throws NotEnoughBalanceException {
+        BlankUser user = new BlankUser();
+        user.setBalance(50);
+        user.setId(1l);
+
+        ShopItem item = itemShopConfig.getShopItem(NONE_BOUGHT_SHOP_ID).get();
+
+        ItemBuyStatus status = shopService.buyItem(user, item);
+        assertThat(status).isEqualTo(ItemBuyStatus.NOT_ENOUGH_MONEY);
+
+        verify(blankUserService, never())
+            .decreaseUserBalance(user, item.getPrice());
+        verify(inventoryService, never()).giveItem(user, item.getItemId(), 1);
+
+        assertThat(buyLogDao.sumOfBoughtItems(item.getId())).isZero();
+    }
+
+    @Test
+    void testBuyItemNoSupply() throws NotEnoughBalanceException {
```

Comment:
what happens when buying multiples and there is existing supply but not as much as requested? Can we add a test for that?

Reply:
I see that it is the same as no supply, no need for an additional test at this time.